html/webappapis/dynamic-markup-insertion/opening-the-input-stream/tasks.window.js

```diff
@@ -0,0 +1,114 @@
+// has been proposed to be removed. This series of tests ensures that this step
+// is no longer executed.
+//
+// This file comprehensively (but not exhaustively) tests for many queued tasks
+// that may be observable. Each taskTest() call in fact runs two tests: the
+// first one "tasks without document.open()" does not actually run
+// document.open(), just to test that the tested task works ordinarily; the
+// second actually calls document.open() to test if the method call removes
+// that specific task from the queue.
+
+setup({
+  allow_uncaught_exception: true
+});
+
+function taskTest(description, testBody) {
+  async_test(t => {
+    const frame = document.body.appendChild(document.createElement("iframe"));
+    // The empty HTML seems to be necessary to cajole Chrome into firing a load
+    // event, which is necessary to make sure the frame's document doesn't have
+    // a parser associated with it.
+    frame.src = "/common/blank.html";
+    t.add_cleanup(() => frame.remove());
+    frame.onload = t.step_func(() => {
+      // Make sure there is no parser. Firefox seems to have an additional
+      // non-spec-compliant readiness state "uninitialized", so test for the
+      // two known valid readiness states instead.
+      assert_in_array(frame.contentDocument.readyState, ["interactive", "complete"]);
+      testBody(t, frame, doc => {});
+    });
+  }, `tasks without document.open() (${description})`);
+
+  async_test(t => {
+    const frame = document.body.appendChild(document.createElement("iframe"));
+    // The empty HTML seems to be necessary to cajole Chrome into firing a load
+    // event, which is necessary to make sure the frame's document doesn't have
+    // a parser associated with it.
+    frame.src = "/common/blank.html";
+    t.add_cleanup(() => frame.remove());
+    frame.onload = t.step_func(() => {
+      // Make sure there is no parser. Firefox seems to have an additional
+      // non-spec-compliant readiness state "uninitialized", so test for the
+      // two known valid readiness states instead.
+      assert_in_array(frame.contentDocument.readyState, ["interactive", "complete"]);
+      testBody(t, frame, doc => doc.open());
+    });
+  }, `document.open() and tasks (${description})`);
+}
+
+taskTest("timeout", (t, frame, open) => {
+  let happened = false;
+  // Work around the lint script, since we can't use step_timeout here.
+  frame.contentWindow['setTimeout'](() => happened = true, 100);
+  open(frame.contentDocument);
+  t.step_timeout(() => {
+    assert_true(happened);
+    t.done();
+  }, 200);
+});
+
+taskTest("window message", (t, frame, open) => {
+  let counter = 0;
+  frame.contentWindow.postMessage(undefined, "*");
+  open(frame.contentDocument);
+  frame.contentWindow.postMessage(undefined, "*");
+  frame.contentWindow.onmessage = t.step_func(e => {
+    assert_equals(e.data, undefined);
+    counter++;
+    assert_less_than_equal(counter, 2);
+    if (counter == 2) {
+      t.done();
```

Comment:
(This is the test that will time out after 30s or so.)

Reply:
This didn't seem to timeout in the latest Travis run…

Reply:
Travis unfortunately doesn't fail unless the tests are flaky, it doesn't help catch any other problem, even totally broken tests. Did you read the logs to see that it didn't time out, or how could you tell?

Reply:
Oh right, I guess I meant rather that it wasn't flaky.